contracts/registry/ReleaseManager.sol

```diff
@@ -0,0 +1,382 @@
+        AccessManaged(accessManager.authority())
+    {
+        require(address(accessManager) > address(0));
+        require(initialVersion.toInt() > 0);
+
+        _accessManager = accessManager;
+
+        _initial = initialVersion;
+
+        _registry = new Registry();
+    }
+
+    // TODO deploy proxy and initialize with given implementation instead of using given proxy?
+    // IMPORTANT: MUST never be possible to create with access/release manager, token registry
+    function createNextRelease(IRegistryService registryService)
+        external
+        restricted // GIF_ADMIN_ROLE
+        returns(NftId nftId)
+    {
+        if(!registryService.supportsInterface(type(IRegistryService).interfaceId)) {
+            revert NotRegistryService();
+        }
+        // TODO unreliable! MUST guarantee the same authority -> how?
+        if(registryService.authority() != authority()) {
+            revert UnexpectedServiceAuthority(
+                authority(), 
+                registryService.authority()); 
+        }
+
+        (
+            IRegistry.ObjectInfo memory info, 
+            bytes memory data
+        ) = _getAndVerifyContractInfo(registryService, SERVICE(), msg.sender);
+
+        VersionPart nextVersion = getNextVersion();
+        _verifyServiceInfo(info, nextVersion, SERVICE());
+
+        _verifyAndStoreConfig(data);
+
+        //setTargetClosed(newRegistryService, true);
+
+        _registerService(address(registryService), nextVersion, SERVICE());
+        
+        nftId = _registry.registerService(info);
+
+        // external call
+        registryService.linkToRegisteredNftId();
+
+        emit LogReleaseCreation(nextVersion, registryService); 
+    }
+
+    function registerService(IService service) 
+        external
+        restricted // GIF_MANAGER_ROLE
+        returns(NftId nftId)
+    {
+        if(!service.supportsInterface(type(IService).interfaceId)) {
+            revert NotService();
+        }
+
+        (
+            IRegistry.ObjectInfo memory info, 
+            //bytes memory data
+        ) = _getAndVerifyContractInfo(service, SERVICE(), msg.sender);
+
+        VersionPart nextVersion = getNextVersion();
+        ObjectType serviceDomain = _verifyServiceInfo(info, nextVersion, zeroObjectType());
+
+        bytes4[] memory selector = _config[nextVersion][serviceDomain].selector;
+
+        // service type is in release
+        if(selector.length == 0) {
+            revert ServiceNotInRelease(service, serviceDomain); 
+        }
+
+        // setup and grant unique role
+        address registryService = getService(nextVersion, SERVICE());
+        RoleId roleId = _accessManager.setAndGrantUniqueRole(
+            address(service), 
+            registryService, 
+            selector);
+
+        _config[nextVersion][serviceDomain].roleId = roleId;
+        _awaitingRegistration--;
+
+        // activate release
+        if(_awaitingRegistration == 0) {
```

Comment:
Since registration of registry service the only way its interface(~set of domains) can change is through proxy upgrade. If upgrade can happen before activation then we need account for addition/deletion of domains

Reply:
Ok, so if i understand this right, then in case we introduce a new service (e.g. the old one got too large again and we need to split) we can upgrade the RegistryService, add the necessary information in the `_config` data and once we register the new RegistryService, we're good to go. Correct? 
I think i wasn't fully aware where the `_config` came from.

Reply:
Yes. I won't call it a registration, because only _config will be updated. But this feature is not there yet.